docker/puppetserver-standalone/docker-entrypoint.d/80-ca.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+
+master_running() {
+  status=$(curl --silent --fail --insecure "https://${CA_HOSTNAME}:8140/status/v1/simple")
+  test "$status" = "running"
+}
+
+hocon() {
+  /opt/puppetlabs/puppet/lib/ruby/vendor_gems/bin/hocon "$@"
+}
+
+CA_HOSTNAME="${CA_HOSTNAME:-puppet}"
+
+if [[ "$ENABLE_CA" != "true" ]]; then
+  # we are just an ordinary compiler
+  echo "turning off CA"
+  cat > /etc/puppetlabs/puppetserver/services.d/ca.cfg <<EOF
+puppetlabs.services.ca.certificate-authority-disabled-service/certificate-authority-disabled-service
+puppetlabs.trapperkeeper.services.watcher.filesystem-watch-service/filesystem-watch-service
+EOF
+
+  ssl_cert=$(puppet master --configprint hostcert)
+  ssl_key=$(puppet master --configprint hostprivkey)
+  ssl_ca_cert=$(puppet master --configprint localcacert)
+  certname=$(puppet config print certname)
+
+
+  cd /etc/puppetlabs/puppetserver/conf.d/
+  hocon -f webserver.conf set webserver.ssl-cert $ssl_cert
+  hocon -f webserver.conf set webserver.ssl-key $ssl_key
+  hocon -f webserver.conf set webserver.ssl-ca-cert $ssl_ca_cert
+  hocon -f webserver.conf set webserver.ssl-crl-path /etc/puppetlabs/puppet/ssl/crl.pem
+  cd /
+
+  # bootstrap certs for the puppetserver
+  if [[ ! -f "/etc/puppetlabs/puppet/ssl/certs/${certname}.pem" ]]; then
+    while ! master_running; do
+      sleep 1
+    done
+
+    puppet agent --noop --server="${CA_HOSTNAME}"
```

Comment:
yeah, this is just for getting ssl certs.

I took a look at the puppet ssl subcommand but it looks like that doesn't exist yet in 5.3.x

Reply:
@underscorgan Do you have any sense if we should use the new `ssl.sh` script instead of the agent run here?

Puppetserver is the one of the containers that actually *will* have a puppet-agent available on it, but wondering if there's value in consistency across our containers and to prevent this agent run from organically growing other responsibilities in the future.

Reply:
I don't have super strong feelings, though leaning slightly towards using ssl.sh for consistency.